spec.html

```diff
@@ -11952,13 +11952,15 @@ <h1>Runtime Semantics: ArgumentListEvaluation</h1>
           1. Let _siteObj_ be GetTemplateObject(_templateLiteral_).
           1. Return a List containing the one element which is _siteObj_.
         </emu-alg>
-        <emu-grammar>SubstitutionTemplate : TemplateHead Expression TemplateSpans</emu-grammar>
+        <emu-grammar>TemplateLiteral : SubstitutionTemplate</emu-grammar>
         <emu-alg>
           1. Let _templateLiteral_ be this |TemplateLiteral|.
           1. Let _siteObj_ be GetTemplateObject(_templateLiteral_).
-          1. Let _firstSubRef_ be the result of evaluating |Expression|.
+          1. Let _first_ be the |Expression| of |SubstitutionTemplate|.
```

Comment:
The pattern `|symbol| of |otherSymbol|` is currently only used in Annex B, I believe, and I'm concerned about encouraging it.

Is there another way to fix this bug that doesn't involve directly digging into the expansion of `SubstitutionTemplate`?

What do you think of defining `ArgumentListEvaluation` for *SubstitutionTemplate* as before, but where it doesn't return *siteObj*, and then define `ArgumentListEvaluation` for *TemplateLiteral : SubstitutionTemplate* which prepends the *siteObj* to that list?

Reply:
> The pattern `|symbol| of |otherSymbol|` is currently only used in Annex B, I believe, ...

It's used a few times outside of Annex B:

Names and Keywords:
    `the |CodePoint| of the |UnicodeEscapeSequence|`
Async Function Definitions:
    `the |FormalParameters| of an |AsyncFunctionDeclaration|, ...`
    `the |BindingIdentifier| of a |FunctionExpression|, ...`
Tail Position Calls:
    `the |FunctionBody| of a |GeneratorBody|`
    `the |FunctionBody| of an |AsyncFunctionBody|`
    `the |FunctionBody| of an |AsyncGeneratorBody|`
Source Text Module Records:
    `the |ModuleSpecifier| of the |ImportDeclaration|`
    `the |ModuleSpecifier| of the |ExportDeclaration|`
Static Semantics: Early Errors (Patterns):
    `the |RegExpIdentifierName| of this production's |GroupName|`

(Not arguing for or against, just supplying info.)

Reply:
> What do you think of defining `ArgumentListEvaluation` for *SubstitutionTemplate* as before, but where it doesn't return *siteObj*, and then define `ArgumentListEvaluation` for *TemplateLiteral : SubstitutionTemplate* which prepends the *siteObj* to that list?

That's a fair approach, though as @jmdyck confirms I got the pattern from other places.